### ISSUE TITLE
Nix flake not building

### ISSUE THREAD

**Issue by purcell** [ledger/ledger#2213]
For the Emacs `ledger-mode` CI we use Nix and are [trying to use](https://github.com/ledger/ledger-mode/pull/354) the `flake.nix` definition here to build the latest development version of `ledger`. However, Nix runs the regression tests at installation time and those tests currently fail. Since the CI for this repo is currently passing, I'm wondering if anyone can guess what the issue might be? (Screenshot of failure below, since links to Action logs expire after a while.)

<img width="886" alt="Screenshot 2023-03-14 at 08 43 03" src="https://user-images.githubusercontent.com/5636/224929726-358fe5df-e9ad-4b95-a549-07ee3ecaf889.png">

**Comment by afh** [comment]
Thanks for raising this, @purcell.

Unfortunately `RegressTest_1057` passes locally for me on the current state of the `master` branch (`8f734a92`) and I assume it passes for you locally too. Looking at the test nothing comes to mind immediately as a cause for the test failure.

To get more context about failures ledger's worklow reruns `ctest` with the following options `--rerun-failed --output-on-failure` (see ledger's [`cmake.yml`](https://github.com/ledger/ledger/blob/master/.github/workflows/cmake.yml#L75)).

Would it be possible to point the [`ledger_version`](https://github.com/ledger/ledger-mode/blob/7d8aa094753184d60e840fc08639803c4e05355c/.github/workflows/test.yml#L20) in your `test.yml` to a ledger fork that has the ctest options `--rerun-failed --output-on-failure` added to the `ctest` command in the `checkPhase` of [ledger's flake.nix](https://github.com/ledger/ledger/blob/master/flake.nix#L55) to see the test result?

In regards to expiring Actions logs I'd like to mention the possibility to view and save/download raw logs of GitHub Actions which can then be attached to issues and provide a more complete picture of a action run and without expiring.

<img width="864" alt="rawlogs" src="https://user-images.githubusercontent.com/16507/224940734-2540ea81-3db5-48b2-935e-a0fd8a2e96c1.png">

Are you okay with me taking inspiration from your work to possibly alter ledger's GitHub action to use Nix too?

**Comment by purcell** [comment]
> Would it be possible to point the [ledger_version](https://github.com/ledger/ledger-mode/blob/7d8aa094753184d60e840fc08639803c4e05355c/.github/workflows/test.yml#L20) in your test.yml to a ledger fork that has the ctest options --rerun-failed --output-on-failure added to the ctest command in the checkPhase of [ledger's flake.nix](https://github.com/ledger/ledger/blob/master/flake.nix#L55) to see the test result?

Trying that now...

> Are you okay with me taking inspiration from your work to possibly alter ledger's GitHub action to use Nix too?

Yes, of course. I think it would be useful to have a CI job here that at least ensures the flake stays working.

**Comment by purcell** [comment]
BTW, if you use Nix in CI here, perhaps add a Cachix so that any build artifacts will be cached, and can perhaps be re-used downstream (e.g. in the `ledger-mode` CI 😁).

**Comment by purcell** [comment]
That still failed, but the output is the same, and not quite sure how to get in there on the Actions runner and see the underlying logs from the build using `nix log`. When I build from the flake locally (`nix build github:ledger/ledger`) it's also fine. Hmmm...

**Comment by akirak** [comment]
@purcell You can run `nix` commands with `--print-build-logs` flag, e.g.:

```sh
nix shell github:purcell/ledger#ledger --print-build-logs -c make -C test test-batch
```

You'll get the following error message:

> ledger> 235/419 Test #237: RegressTest_1057 ......................................***Failed    0.07 sec
ledger> FAILURE in output from /build/source/test/regress/1057.test:
ledger> --
ledger> $ledger -f "/build/source/test/regress/1057.test" --now=2014/06/27 emacs
ledger> --
ledger>   @@ -1,4 +1,4 @@
ledger>   -(("/build/source/test/regress/1057.test" 1 (21308 60112 0) nil "www.amazon.fr"
ledger>   +(("/build/source/test/regress/1057.test" 1 (21308 42112 0) nil "www.amazon.fr"
ledger>      (2 "Dépense:Loisir:Ordi:Matériel" "101,50 €" nil " disque dur portable 2,5\" 2000 Go")
ledger>      (3 "Dépense:Maison:Service:Poste" "70,13 €" nil)
ledger>      (4 "Passif:Crédit:BanqueAccord" "-171,63 €" t)))
ledger> E[1057.test]STDERR:
ledger> b''
ledger> FAILED (1)

**Comment by purcell** [comment]
Doh, thanks, I'll give that a go!

**Comment by tbm** [comment]
that error suggests the timezone variable isn't properly set, I believe.

See `CMakeLists.txt`: `set(Ledger_TEST_TIMEZONE "America/Chicago")`

**Comment by purcell** [comment]
Hmmm, interesting! I'm not familiar enough with `cmake`/`ctest` to see why that might not be taking effect.

**Comment by tbm** [comment]
I might be completely wrong but a random though: maybe this happens if the timezone database is not installed (`tzdata` on Debian, not sure about Nix).

Again, I might be completely wrong.

**Comment by purcell** [comment]
Indeed! Adding `tzdata` to the `nativeBuildInputs` seems to fix this.

**Comment by purcell** [comment]
See https://github.com/purcell/ledger/commit/1dcb4978b1e4953190f0799264df20d6ebecf55c

**Comment by tbm** [comment]
I think you have write permission to the ledger repo. if not, please open a PR

**Comment by purcell** [comment]
Yes, true, I'll fix it, perhaps tomorrow.

**Comment by afh** [comment]
Happy to take it off your plate, if you like, @purcell…

**Comment by purcell** [comment]
`master` is apparently protected and I can't merge a PR without a review, so the fix is in #2216 if you'd like to check it over. Thanks!
